File: src/problem036/solution036.erl
-module(solution036).
-export([start/0]).

start() ->
    io:format("~p\n", [lists:foldl(fun(X, Y) -> X + Y end, 0,
        lists:filter(fun(X) ->
                isPalindrome("~.2B", X) and isPalindrome("~.10B", X) end,
            lists:seq(1, 1000000)))]).

isPalindrome(FormatStr, Num) ->
    NumStr = hd(io_lib:format(FormatStr, [Num])),
    (NumStr == lists:reverse(NumStr)).

File: src/problem036/solution036.hs
import Data.Char (intToDigit)
import Numeric (showIntAtBase)
main = print . sum . filter (isPalindrome 2) . filter (isPalindrome 10) $
       [1..999999]
isPalindrome base x = x' == reverse x'
    where x' = showIntAtBase base intToDigit x ""
